### PR TITLE
Writing a line containing 'split' and 'strip' being used at the same …

### PR DESCRIPTION
**Pull request by mamkhari**
…time into two separate lines: first split the string then strip it in the next line.

### CODE REVIEW

File: extract_unique_sensors.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 for sensor in candidate_lines:
     # Typical output from "sensor" variable:
     # kat.sensors.anc_api_version   kat.sensors.anc_ganglia_api_version   kat.sensors.anc_ganglia_kat_monctl_system_os_name
-    sensor = sensor.split('.')[-1].strip("\n")
+    sensor = sensor.split('.')[-1] # to split a string using a 'dot'character.
+    sensor.strip("\n") # to remove newlines in a string
```
Comment (mmphego) [inline extract_unique_sensors.py]: This has no effect, you are merely stripping of `\n` and not overwriting `sensor` variable.

This is what you should do instead:
`sensor = sensor.strip("\n")`

File: extract_unique_sensors.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 for sensor in candidate_lines:
     # Typical output from "sensor" variable:
     # kat.sensors.anc_api_version   kat.sensors.anc_ganglia_api_version   kat.sensors.anc_ganglia_kat_monctl_system_os_name
-    sensor = sensor.split('.')[-1].strip("\n")
+    sensor = sensor.split('.')[-1] # to split a string using a 'dot'character.
```
Comment (mmphego) [inline extract_unique_sensors.py]: There should be 2 spaces before the beginning of the comment.

**Review by mmphego — CHANGES_REQUESTED** [review]
Make the fixes as explained.

File: extract_unique_sensors.py [hunk 3]
```diff
@@ -10,11 +10,12 @@
...
 
-# Loop over candidate_lines and extract sensor names
+#  Loop over candidate_lines and extract sensor names
 sensor_names = []
 for sensor in candidate_lines:
-    # Typical output from "sensor" variable:
-    # kat.sensors.anc_api_version   kat.sensors.anc_ganglia_api_version   kat.sensors.anc_ganglia_kat_monctl_system_os_name
-    sensor = sensor.split('.')[-1].strip("\n")
+#  Typical output from "sensor" variable:
+#  kat.sensors.anc_api_version   kat.sensors.anc_ganglia_api_version   kat.sensors.anc_ganglia_kat_monctl_system_os_name
```
Comment (mmphego) [inline extract_unique_sensors.py]: wrt to spaces after `#`, please familiarize yourself with https://www.python.org/dev/peps/pep-0008/#id31

This is the sacred [style guide for Python](https://www.python.org/dev/peps/pep-0008), know it and breath it :laughing:

File: extract_unique_sensors.py [hunk 4]
```diff
@@ -10,11 +10,12 @@
...
+    sensor = sensor.split('.')[-1]  # to split a string using a 'dot'character.
+    sensor = sensor.strip("\n")  # to remove newlines in a string
```
Comment (mmphego) [inline extract_unique_sensors.py]: Great improvement.

File: extract_unique_sensors.py [hunk 5]
```diff
@@ -10,11 +10,12 @@
         for candidate_line in line.split():
             candidate_lines.append(candidate_line)
 
-# Loop over candidate_lines and extract sensor names
+#  Loop over candidate_lines and extract sensor names
```
Comment (mmphego) [inline extract_unique_sensors.py]: See: https://www.python.org/dev/peps/pep-0008/#id31

**Review by mmphego — CHANGES_REQUESTED** [review]
Almost there...

**Comment by mamkhari** [comment]
> Almost there...

What's next?Merge the pull request?

**Comment by mmphego** [comment]
> > Almost there...
> 
> What's next?Merge the pull request?

Please address the comments.

File: extract_unique_sensors.py (hunk 5 above)
Comment (mmphego) [reply extract_unique_sensors.py]: @mamkhari

File: extract_unique_sensors.py (hunk 3 above)
Comment (mmphego) [reply extract_unique_sensors.py]: @mamkhari

File: extract_unique_sensors.py [hunk 6]
```diff
@@ -10,12 +10,12 @@
         for candidate_line in line.split():
             candidate_lines.append(candidate_line)
 
-#  Loop over candidate_lines and extract sensor names
+  #  Loop over candidate_lines and extract sensor names
```
Comment (mmphego) [inline extract_unique_sensors.py]: This is how it should be written.
```suggestion
# Loop over candidate_lines and extract sensor names 
```

File: extract_unique_sensors.py [hunk 7]
```diff
@@ -10,12 +10,12 @@
...
 sensor_names = []
 for sensor in candidate_lines:
-#  Typical output from "sensor" variable:
-#  kat.sensors.anc_api_version   kat.sensors.anc_ganglia_api_version   kat.sensors.anc_ganglia_kat_monctl_system_os_name
-    sensor = sensor.split('.')[-1]  # to split a string using a 'dot'character.
-    sensor = sensor.strip("\n")  # to remove newlines in a string
+  #  Typical output from "sensor" variable:
+  #  kat.sensors.anc_api_version   kat.sensors.anc_ganglia_api_version   kat.sensors.anc_ganglia_kat_monctl_system_os_name
```
Comment (mmphego) [inline extract_unique_sensors.py]:
```suggestion
    # Typical output from "sensor" variable:
    # kat.sensors.anc_api_version   kat.sensors.anc_ganglia_api_version   kat.sensors.anc_ganglia_kat_monctl_system_os_name
```

**Review by mmphego — CHANGES_REQUESTED** [review]

**Review by mmphego — APPROVED** [review]

**Pull request merged**